tests/test_insert.py

```diff
@@ -147,3 +147,57 @@ def test_insert_return(self):
                 'INSERT INTO test (a) VALUES', [(x,) for x in range(5)]
             )
             self.assertEqual(rv, 5)
+
+
+class InsertColumnarTestCase(BaseTestCase):
+    def test_insert_tuple_ok(self):
+        with self.create_table('a Int8, b Int8'):
+            data = [(1, 2, 3), (4, 5, 6)]
+            self.client.execute(
+                'INSERT INTO test (a, b) VALUES', data, columnar=True
+            )
+
+            query = 'SELECT * FROM test'
+            inserted = self.emit_cli(query)
+            self.assertEqual(inserted, '1\t4\n2\t5\n3\t6\n')
+            inserted = self.client.execute(query)
+            self.assertEqual(inserted, [(1, 4), (2, 5), (3, 6)])
+            inserted = self.client.execute(query, columnar=True)
+            self.assertEqual(inserted, [(1, 2, 3), (4, 5, 6)])
+
+    def test_insert_data_different_column_length(self):
+        with self.create_table('a Int8, b Int8'):
+            with self.assertRaises(ValueError) as e:
+                data = [(1, 2, 3), (4, 5)]
+                self.client.execute(
+                    'INSERT INTO test (a, b) VALUES', data, columnar=True
+                )
+            self.assertEqual(str(e.exception), 'Expected 3 rows, got 2')
+
+    def test_data_invalid_types(self):
+        with self.create_table('a Int8'):
+            with self.assertRaises(TypeError) as e:
+                data = [{'a': 1}, (2, )]
+                self.client.execute(
+                    'INSERT INTO test (a) VALUES', data,
+                    types_check=True, columnar=True
+                )
+
+            self.assertIn('list or tuple is expected', str(e.exception))
```

Comment:
As for now, the only valid format for columnar data is `list` of `tuples` where each `tuple` represents all values of particular column, e.g. `[(1,2,3), ('a', 'b', 'c')]`. I just wanted to show that all list items will be checked, not only the first one.

I think the valid `dict` format for columnar data will be something like this:
```
{
    'a': [1,2,3],
    'b': ['a', 'b', 'c']
}
```
I can support this format as well if you found it useful. It would be great if you have a suggestion about what's the best way to distinguish insert data from query params.